src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -535,12 +535,43 @@ using stan::model::index_omni;
 using stan::model::nil_index_list;
 using namespace stan::math; |}
 
+let pre_boilerplate =
+  {|#include <vector>
+#include <Eigen/Dense>
+
+template <typename T, typename S>
+std::vector<T> resize_to_match(std::vector<T>& dst, const std::vector<S>& src) {
+  dst.resize(src.size());
+  return dst;
```

Comment:
does the old compiler do something like this too?

Reply:
No, these are new helpers for emitting unsized declarations. I used to need that in this PR but now we don't, but I figured it was helpful to have some code that works for that use case in there even if nothing is generating unsized declarations yet?